### ISSUE TITLE
ECHConfig test cipher suites

### ISSUE THREAD

**Issue by sayrer** [cloudflare/go#76]
It looks like the test code is exercising fewer HPKE cipher suites than it used to, and the change happened during the upgrade from draft-09 to draft-10. Perhaps the code in this repo should also assert on the length of the cipher suites list. I think it just loops.

Noticed this while working on Rustls:
https://github.com/grafica/rustls/blob/cde5c28faa03155350f51c49b8c6438391fc28bc/rustls/src/ech_key.rs#L189

**Comment by cjpatton** [comment]
Hi @sayrer!

>  Perhaps the code in this repo should also assert on the length of the cipher suites list. I think it just loops.

Could you clarify what you mean by "assert on the length of the cipher suites list"?

**Comment by sayrer** [comment]
I don't think this function in `test_ech.go`:

```
func TestUnmarshalConfigs(t *testing.T) {
	block, rest := pem.Decode([]byte(echTestConfigs)
```

or any related ones, check the number of cipher suites that should be found (which is dependent on the PEM in the test file). This wouldn't exactly test the correctness of the decoding, but it would raise an eyebrow if the number in that assertion went down (exactly what happened in my patch).

**Comment by cjpatton** [comment]
Good idea. What do you think of this? https://github.com/cloudflare/go/pull/77

**Comment by sayrer** [comment]
I think that's the right approach, but shouldn't there be 4 cipher suites, as there were in the draft-09 PEM? (maybe you are planning to fix this elsewhere, or it's intentional)

**Comment by sayrer** [comment]
> I think that's the right approach, but shouldn't there be 4 cipher suites, as there were in the draft-09 PEM? (maybe you are planning to fix this elsewhere, or it's intentional).

[edit: Additionally, I think the test config used to have a few different test KEM types: `DHKEM_P256_HKDF_SHA256` and `DHKEM_X25519_HKDF_SHA256`.  They're both x25519 now.]

**Comment by cjpatton** [comment]
Right, we used to have more cipher suites in the test data.
- The reason we had two KDFs was that, prior to ECH-10, the config_id depended on the KDF. Thus, we needed to exercise the case in which each config might have multiple IDs. Since ECH-10, that's no longer the case, so we removed the second KDF. 
- We dropped the P256 KEM simply because we aren't using it for crypto.cloudflare.com. Though if it would help, I'd be happy to generate it for the tests.
